services/FacilityService.php

```diff
@@ -0,0 +1,283 @@
+        $sql .= "     street='" . $data["street"] . "',";
+        $sql .= "     city='" . $data["city"] . "',";
+        $sql .= "     state='" . $data["state"] . "',";
+        $sql .= "     postal_code='" . $data["postal_code"] . "',";
+        $sql .= "     country_code='" . $data["country_code"] . "',";
+        $sql .= "     federal_ein='" . $data["federal_ein"] . "',";
+        $sql .= "     website='" . $data["website"] . "',";
+        $sql .= "     email='" . $data["email"] . "',";
+        $sql .= "     color='" . $data["color"] . "',";
+        $sql .= "     service_location='" . $data["service_location"] . "',";
+        $sql .= "     billing_location='" . $data["billing_location"] . "',";
+        $sql .= "     accepts_assignment='" . $data["accepts_assignment"] . "',";
+        $sql .= "     pos_code='" . $data["pos_code"] . "',";
+        $sql .= "     domain_identifier='" . $data["domain_identifier"] . "',";
+        $sql .= "     attn='" . $data["attn"] . "',";
+        $sql .= "     tax_id_type='" . $data["tax_id_type"] . "',";
+        $sql .= "     primary_business_entity='" . $data["primary_business_entity"] . "',";
+        $sql .= "     facility_npi='" . $data["facility_npi"] . "',";
+        $sql .= "     facility_code='" . $data["facility_code"] . "'";
+
+        return sqlInsert($sql);
+    }
+
+    /**
+     * Shared getter for the various specific facility getters.
+     *
+     * @param $map - Query information.
+     * @return array of associative arrays | one associative array.
+     */
+    private function get($map)
+    {
+        $where = isset($map["where"]) ? $map["where"] : null;
+        $data  = isset($map["data"])  ? $map["data"]  : null;
+        $join  = isset($map["join"])  ? $map["join"]  : null;
+        $order = isset($map["order"]) ? $map["order"] : null;
+        $limit = isset($map["limit"]) ? $map["limit"] : null;
+
+        $sql  = " SELECT FAC.id,";
+        $sql .= "        FAC.name,";
+        $sql .= "        FAC.phone,";
+        $sql .= "        FAC.fax,";
+        $sql .= "        FAC.street,";
+        $sql .= "        FAC.city,";
+        $sql .= "        FAC.state,";
+        $sql .= "        FAC.postal_code,";
+        $sql .= "        FAC.country_code,";
+        $sql .= "        FAC.federal_ein,";
+        $sql .= "        FAC.website,";
+        $sql .= "        FAC.email,";
+        $sql .= "        FAC.service_location,";
+        $sql .= "        FAC.billing_location,";
+        $sql .= "        FAC.accepts_assignment,";
+        $sql .= "        FAC.pos_code,";
+        $sql .= "        FAC.x12_sender_id,";
+        $sql .= "        FAC.attn,";
+        $sql .= "        FAC.domain_identifier,";
+        $sql .= "        FAC.facility_npi,";
+        $sql .= "        FAC.tax_id_type,";
+        $sql .= "        FAC.color,";
+        $sql .= "        FAC.primary_business_entity,";
+        $sql .= "        FAC.facility_code,";
+        $sql .= "        FAC.extra_validation";
```

Comment:
I'm not a purist about these things, but I generally follow the wisdom of http://stackoverflow.com/a/3640051/1525534